src/proto/generation.proto

```diff
@@ -219,6 +219,15 @@ message TransformType {
   }
 }
 
+message CAIParameters {
```

Comment:
I tried to make this as extensible as possible, while also avoiding things like booleans (since in the future, we might want to allow users to provide some additional metadata in the form of a string). Not sure if I've over complicated things with the message or not.